### DISCUSSION TITLE
zh/guide/drivers/onedrive_app

### DISCUSSION THREAD

**Discussion by giscus** [AlistGo/docs#189] [Announcements]
# zh/guide/drivers/onedrive_app

好处以及注意事项 1. 好处是只授权一次管理员登录，后续添加参数时用一样的，在添加时除了邮箱哪里不同，别的都一样 例如E5 25个5T的账号，你提前登录初始化完毕后，直接CV之前的参数即可，不用像以前一样每次都得去获取刷新令牌时间一长还失效; A1 A1P等可以分发的账号更多(500 - 无限); 2. 这样的方式管理员授权后，可以随意查看组织中每个成...

https://alist.nn.ci/zh/guide/drivers/onedrive_app.html

**Comment by ypq123456789** [comment]
可选驱动里面没有onedrive app啊？更新alist也没用。另外第五步后卡在redirecting了

**Reply by anwen-anyi** [reply]
onedrive app下个版本上线，
redirecting报错信息看看你登录的是全局管理员账号吗？
回调参数在注册应用时候写的是什么呢？

**Comment by ypq123456789** [comment]
登录的是全局管理员账号。重定向URL(回调参数)，类型选择Web，参数填写： https://entra.microsoft.com/TokenAuthorize ，否则后面授权时无法授权
就是按照这个参数填的。



---Original---
From: ***@***.***&gt;
Date: Fri, Mar 24, 2023 23:31 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [alist-org/docs] zh/guide/drivers/onedrive_app (Discussion #189)




 
onedrive app下个版本上线，
 redirecting报错信息看看你登录的是全局管理员账号吗？
 回调参数在注册应用时候写的是什么呢？

**Reply by anwen-anyi** [reply]
你指的是不是这样？ 
按照 `https://login.microsoftonline.com/{租户ID或者common}/adminConsent?client_id={客户端ID}&redirect_uri={注册应用时的回调URL}` 这个格式访问点击授权后，页面变成这样了？   
![image](https://user-images.githubusercontent.com/56105412/227680416-1a907ed2-22c8-4295-be34-cccfffb0d4e2.png)

**Reply by 2-3-5-7** [reply]
大家注意是“全局管理员账号”，我第一次登录的是注册 E5 的账号，就弄错了

**Comment by ypq123456789** [comment]
是的，就是这样一个页面了。



---Original---
From: ***@***.***&gt;
Date: Sat, Mar 25, 2023 10:09 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [alist-org/docs] zh/guide/drivers/onedrive_app (Discussion #189)




 
你指的是不是这样？
 按照 https://login.microsoftonline.com/{租户ID或者common}/adminConsent?client_id={客户端ID}&amp;redirect_uri={注册应用时的回调URL} 这个格式访问点击授权后，页面变成这样了？

**Reply by anwen-anyi** [reply]
仔细看顶部链接栏哪里 是不是一样的域名? 如果是就是没错

**Reply by ypq123456789** [reply]
是的，所以这样就是代表授权成功了是吧？

**Reply by anwen-anyi** [reply]
是的，你直接去添加就好了没问题了

**Comment by 5315929** [comment]
这里的教程能不能不要只写一半啊？onedrive app的还有api权限，后续授权方面能都贴出来吗？真的是每个教程看了都是一脸懵。百度又找不到其他的教程。每次加某个盘都要折腾半天。你们做alist是为了让大家更方便，次次教程都出一半是几个意思？

**Reply by anwen-anyi** [reply]
?  眼睛不要可以捐了

**Reply by 5315929** [reply]
那个客户端密钥怎么来的？教程里说清楚了？授权教程里也没有，是你自己眼睛有问题吧

**Reply by anwen-anyi** [reply]
你眼瞎？ 看不到第二步？ 你楼上的怎么能获取？

**Reply by 5315929** [reply]
我是小白。第一次接触这个。我是看了评论，才发现还要获得授权，还有第二步填写alist的时候那个客户端密钥也不知道去哪里获得。获得授权应该是最后一步了吧？这个教程里也没写出来，教程只有到注册应用这一步，获取了一个客户端id。然后剩下的就没有了，直接就是填写示例了。你能清理下你的浏览器缓存，在来看看嘛？教程真的只有一半

**Reply by anwen-anyi** [reply]
这么大的提示没有看吗？ :thinking: :thinking: :thinking:
小白不是你的理由 好好说话，不欠你的。
![image](https://user-images.githubusercontent.com/56105412/227756047-53c47779-a9eb-48ca-91ac-84c935e3c40e.png)

**Reply by 5315929** [reply]
好吧。我还真第一次见过这样也能设置翻页的。大佬，失敬！！确实没看到这一行翻页小字。一半都是文章末尾才有

**Comment by 1hinhin** [comment]
哭了 第4步不能代表管理员同意，你们是怎么做到的，是学校原因吗

**Reply by anwen-anyi** [reply]
可能吧 你不是admin？ 没那么多环境测试 测试的时候只拿了E5深度测试了

**Comment by larkhover** [comment]
请问这种方式支持挂载sharepoint吗

**Comment by xiaopaul** [comment]
关于批量添加Onedrive账号的教程有些不太清楚，请教大佬。
1，我的Alist装在群晖上，怎么将Alist_patch做好的配置传到群晖上？或者可以直接在群晖进行配置运行？这里不是特别了解。
2，我有多个E5账号，同时也是管理员。是否只需要获取其中一个账号的APP ID，Tenant ID和 Secret并进行管理员授权，就可以批量添加其他同一个域的账号，而不需要逐一寻找其他账号的AppID和Secret。

非常感谢。

**Reply by xiaopaul** [reply]
或者其他账号都不需要再去设置APP了？

**Reply by anwen-anyi** [reply]
直接在群辉配置不太懂 应该可以吧 只要哪个批量运行的软件有群辉能用的版本~
是的 只需要获取管理员的客户端ID和秘钥就行，然后手动把同域内的其他账号的账户导出来自己 批量添加就行

**Reply by xiaopaul** [reply]
收到，非常感谢。我一个个填上去算了，也不会太耗时间，比Rclone 方便很多，就是感觉速度似乎差一点，但不会太影响。
再次感谢团队。

**Comment by love-Liushuang** [comment]
全部流程走完 ，管理员账号加载正常，其他账号加载报错提示“failed get objs: failed to list objs: Unable to retrieve user's mysite URL.” 是什么原因呢， 报错账号的onedrive已经开启状态。这是什么原因啊？

**Reply by anwen-anyi** [reply]
:joy_cat: 此E5账号下面所有的用户（管理员账号除外）都这样吗？

**Reply by love-Liushuang** [reply]
不是E5 就是商业版用户，子账号开通onedrive后不是实时生效 我等了一天多咱们alist才正常，可能微软授权的问题，现在已经好了，子账号开通时效都不一样 有的快有的慢

**Reply by anwen-anyi** [reply]
好吧，可能当场开通会延迟生效（我测试的账号都是以前已经提前开好的） 解决了就行

**Reply by ohmybobo** [reply]
![image](https://github.com/alist-org/docs/assets/30516490/ce85e8c9-52f7-45e7-9dd3-bd2269a1ac32)

登录onedrive网页，点击“返回到经典Onedrive”，可以看到图片上的解除，点击后即可用Alist打开网盘。

**Reply by youyou-sudo** [reply]
## ☝另外一种情况
当你添加的 Onedrive app 存储在某一天突然更改了邮箱，或创建存储时错填邮箱为其他邮箱并保存后，又进入编辑为正确邮箱时，也会出现这样的报错

## ⚙解决方案

1. 复制租户 ID 等配置到便签
2. 删除这个报错目录存储并重新创建新的存储
3. 填入便签中正确配置和**正确邮箱**
（如果填错邮箱并保存请直接删除存储重新创建填写）

## 💡思路
每个添加的存储都有一个 ID，删除并重建存储来达到更换 ID 值，使得 alist 使用新的 ID 标识的缓存，问题解决
> 我使用的alist版本：v3.29.1

![image](https://github.com/alist-org/docs/assets/60727068/a2cafb2e-0206-440f-ad35-fd9b20cca916)

**Reply by gdtv** [reply]
我也遇到这个问题，我的账号创建了几年月的。我的原因是，我创建了账号后从来没有使用过，没有在网页版onedrive上登录过。解决方法是登录一下就行。

**Comment by yizeijun** [comment]
failed get objs: failed to list objs: Either scp or roles claim need to be present in the token.
出现这个代码，但是我在代表管理员同意的时候也完成了，不知道是哪里的问题，主邮箱可以添加，其他邮箱无法添加
![1684472502807.png](https://img1.imgtp.com/2023/05/19/fazepvJ7.png)
但是alist后台显示为work状态，请指教

**Reply by anwen-anyi** [reply]
你授权的怎么和我的不一样？

![image](https://github.com/alist-org/docs/assets/56105412/15a0f68e-bf5c-4301-bced-f30afd10d818)

授权需要在这个路径  不是在当前页面的API授权嗷
左侧侧边栏 ---> 企业应用程序 ---> 所有应用程序 ---> {选择我们开始新建的哪个应用名称} ---> 权限 ---> 点击 代表xxxx授予管理员同意
仔细看下 https://www.bilibili.com/video/BV1Ro4y1s725?t=171.1 视频的说明

感觉你的不对有些

**Reply by yizeijun** [reply]
感谢回复，已完成调整，多谢多谢

**Reply by anwen-anyi** [reply]
是哪里不对吗？ 记录一下 避避坑

**Reply by yizeijun** [reply]
第四步权限那里我搞错了，需要添加程序权限，我是直接添加的。另外使用本地代理可能也会导致错误，建议刚开始使用302重定向，然后成功后再使用本地代理

**Reply by anwen-anyi** [reply]
哈哈 我猜也是，大部分人都是在这里翻车的QAQ

**Reply by Chellane** [reply]
没有找到第四步的企业应用程序导航栏，是不是意味着我的账户不适合这种挂载方式

**Reply by gaojunyang666** [reply]
> 第四步权限那里我搞错了，需要添加程序权限，我是直接添加的。另外使用本地代理可能也会导致错误，建议刚开始使用302重定向，然后成功后再使用本地代理

可以详细说一下吗， 我感觉我每一步都很对，但是就是报错
![image](https://github.com/user-attachments/assets/f683ef0a-094c-46ba-b9d9-f3e8b1e826d1)

**Comment by KimmyLueng** [comment]
按步骤添加了，但网页提示failed get objs: failed to list objs: User's mysite not found.

**Reply by anwen-anyi** [reply]
https://github.com/alist-org/docs/discussions/189#discussioncomment-5928892

**Comment by atomlong** [comment]
离线下载一个476MB的文件，出现这个错误：
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN""http://www.w3.org/TR/html4/strict.dtd"> <HTML><HEAD><TITLE>Service Unavailable</TITLE> <META HTTP-EQUIV="Content-Type" Content="text/html; charset=us-ascii"></HEAD> <BODY><h2>Service Unavailable</h2> <hr><p>HTTP Error 503. The service is unavailable.</p> </BODY></HTML>

**Comment by dan3577** [comment]
最后一步添加一直转圈圈，，返回存储查看状态提示：Post "https://login.microsoftonline.com/我的租户ID/oauth2/token": context deadline exceeded (Client.Timeout exceeded while awaiting headers)

**Reply by anwen-anyi** [reply]
> context deadline exceeded (Client.Timeout exceeded while awaiting headers)

这种一般都是网络问题

**Comment by not-yoursGit** [comment]
最后一步提示这个Failed init storage: Post "https://login.microsoftonline.com/6c89cd46-3c87-4071-81eb-26ba79fcab39/oauth2/token": dial tcp [*******************]:443: connect: network is unreachable怎么办啊，各位大佬？

**Reply by anwen-anyi** [reply]
很明显你网络问题

**Comment by liuhao4399** [comment]
failed get objs: failed to list objs: User not found 请问这是啥原因啊

**Comment by bacon000111** [comment]
Failed update storage in database: UNIQUE constraint failed: x_storages.mount_path
請問這是哪一步有問題?
謝謝

**Reply by Chellane** [reply]
没有找到第四步的企业应用程序导航栏，是不是意味着我的账户不适合这种挂载方式

**Comment by letuouyiya** [comment]
求教。等了1天还是这样，是我的账号有问题嘛？

![Screenshot of a comment on a GitHub issue showing an image, added in the Markdown, of an Octocat smiling and raising a tentacle.](https://raw.githubusercontent.com/letuouyiya/issue/14e3c97b3daf82ca249636404c3f1adb9ae6b97f/%E5%BE%AE%E4%BF%A1%E5%9B%BE%E7%89%87_20231018195934.png)

![Screenshot of a comment on a GitHub issue showing an image, added in the Markdown, of an Octocat smiling and raising a tentacle.](https://raw.githubusercontent.com/letuouyiya/issue/14e3c97b3daf82ca249636404c3f1adb9ae6b97f/%E5%BE%AE%E4%BF%A1%E5%9B%BE%E7%89%87_20231018202259.png)

**Comment by youlan2333** [comment]
求助：全部流程走完 ，管理员账号加载报错提示"failed get objs: failed to list objs: The resource could not be found."，请问如何解决？

**Reply by Maskatise** [reply]
我也是这个问题，请问解决了吗

**Reply by xuanxuan1231** [reply]
里面没东西吧……

**Comment by hide-on-bush-lalala** [comment]
碰到了一个诡异的问题，为啥最后链接无法登录。

![4a51f2f55fc853dfa0c2bac34213a935.png](https://i.mji.rip/2023/10/25/4a51f2f55fc853dfa0c2bac34213a935.png)

**Reply by hide-on-bush-lalala** [reply]
链接就是教程里的链接：https://entra.microsoft.com/TokenAuthorize，访问的链接格式也没问题啊

**Reply by hide-on-bush-lalala** [reply]
问题已解决我是瓜皮，各位切记 %redirect_uri={https://entra.microsoft.com/TokenAuthorize} ，这是一个错误示范，请把所有的括号都删了！！！

%redirect_uri=https://entra.microsoft.com/TokenAuthorize

这样才对！虽然是个小问题，搞了半天！前面也要删！

**Comment by aaa264** [comment]
添加的时候显示Failed init storage: AADSTS900023: Specified tenant identifier 'c47f1640-2749-4883-8d57-d6c9091e64' is neither a valid DNS name, nor a valid external domain. Trace ID: 9a45d14e-7e19-4b2d-8a29-e468955b3900 Correlation ID: 8c9637e0-8a53-4cc5-89cf-90324478d919 Timestamp: 2023-10-30 13:07:06Z，
请问如何解决？

**Comment by EX-XD** [comment]
最后一步访问网址【https://login.microsoftonline.com/{租户ID或者common}/adminConsent?client_id={客户端ID}&redirect_uri={注册应用时的回调URL}】授权时能跳转到教程写的网址，但不是空白页，而是在等待很久后提示需要更换浏览器……
【Try that again using a different browser
We couldn't connect to that service, likely because of settings put in place by your IT team. Open Azure in a different Web browser to try again.】
我的是Microsoft Edge，为了装插件稍微改了管理设置，其他没动，这个也会有影响？

**Reply by EX-XD** [reply]
时隔三天瞟了一眼AList上的网盘，已经能刷出文件了，当时是刷不出的……呃，所以大概是报错也没关系，仅供参考。

**Comment by zhouxinhao1208** [comment]
Failed init storage: AADSTS7000215: Invalid client secret provided. Ensure the secret being sent in the request is the client secret value, not the client secret ID, for a secret added to app 'bdb709f6-e832-471f-8d84-cd99be86f81f'. Trace ID: c0a577d5-9d1d-4413-852a-2bbc91fd9c00 Correlation ID: c134b45c-51fb-431a-a744-cd19f2a2fb5d Timestamp: 2023-12-03 07:25:28Z

**Reply by xuanxuan1231** [reply]
Client Secret用值，不是ID

**Comment by locyi** [comment]
我的时个人版，但在最后一步查看组织所有用户时出了问题，You can't sign in here with a personal account. Use your work or school account instead.

**Reply by chengnan049** [reply]
个人版当然没有组织这种说法啊

**Comment by RunlingDev** [comment]
如何挂载指定目录？

**Reply by xuanxuan1231** [reply]
填根文件夹路径

**Comment by my208928** [comment]


**Reply by my208928** [reply]
创建应用后出现权限不足无法访问，我每次进入页面一直弹出
在尝试检索访问令牌时，门户遇到了问题。建议尝试重新登录，或者在没有访问令牌的情况下继续操作，但这可能会导致用户体验不理想。其他详细信息: interaction_required: AADSTS16000: User account '{EUII Hidden}' from identity provider 'live.com' does not exist in tenant 'Microsoft Services' and cannot access the application '74658136-14ec-4630-ad9b-26e160ff0fc6'(ADIbizaUX) in that tenant. The account needs to be added as an external user in the tenant first. Sign out and sign in again with a different Azure Active Directory user account. Trace ID: 8b9c06a3-2aaa-4543-bcfb-86200a983600 Correlation ID: fe79e400-f2f0-4737-bc7b-fca90f06f9f7 Timestamp: 2024-03-18 09:23:38Z。

**Comment by my208928** [comment]
我卡在了侧边栏中没有企业应用程序，请问怎么解决

**Reply by my208928** [reply]
稀里糊涂的解决了，但是我登录了我的onedirve，发现竟然没有文件，但是我在alist中上传爆满5G限制了，我也搞不懂，不是上传到我的账号吗？我在我的账号中空间使用0，可以上传文件，到alist就爆满限制了

**Comment by chibm** [comment]
E5的25个子号的od容量能否合到一个路径下面,这不就相当于有个超大盘了....

**Comment by lanqianlv** [comment]
我是e5开发者的账户，按照教程一步步解决了，感谢开发组。以前只是听说过管理员可以看用户文件但没在意，现在发现这么容易后有点心情复杂，不过以前用别人的A1和E5也没存什么东西不担心。

**Comment by jsnsjxjw** [comment]
请问alist挂载onedrive后显示failed get objs: failed to list objs: User not found咋办啊？
注册过程没啥问题，看着教程试了两次一样，别的云盘倒是可以正常挂载。

**Comment by ChdCZ** [comment]
Post "https://login.microsoftonline.com/6032f9f9-2452-4497-8a9f-113171c873a3/oauth2/token": dial tcp: lookup login.microsoftonline.com on [::1]:53: read udp [::1]:49922->[::1]:53: i/o timeout

**Reply by ChdCZ** [reply]
之前挂载好的，今天登上去发现打不开了，看了下提示这个，请问是什么原因呀，期待回复，感谢。

**Reply by anwen-anyi** [reply]
> 之前挂载好的，今天登上去发现打不开了，看了下提示这个，请问是什么原因呀，期待回复，感谢。

你网络问题

**Comment by XuanMouren2004** [comment]
为什么我的
![Uploading IMG_0232.jpeg…]()
显示work，但是出现了这样的报错，在网上也没找到解决办法failed get objs: failed to list objs: Insufficient privileges to complete the operation.

**Reply by XuanMouren2004** [reply]
failed get objs: failed to list objs: Tenant does not have a SPO license.现在是这样的报错

**Reply by xuanxuan1231** [reply]
[微软官方](https://learn.microsoft.com/en-us/answers/questions/1526573/tenant-does-not-have-a-spo-license)
> Here are the steps to resolve this issue:
> 
> Assign a SharePoint Online License: You need to assign a SharePoint Online license to the user. You can do this by logging into the Office 365 admin center, selecting the user, and checking the “SharePoint Online” box in the “Manage Product Licenses” section.
Use a Work/School Account: Ensure you’re using a work or school account, not a personal Microsoft account. The /tenant id or /contoso.onmicrosoft.com endpoint only allows users with work/school accounts of a specific Azure AD tenant to log in to the application.

**Comment by zhstzzy** [comment]
怎么挂载指定的目录啊，直接填写文件夹路径没有用

**Comment by Recallzh** [comment]
上传文件出现错误是什么情况

**Comment by FLC-ytfl** [comment]
failed get objs: failed to list objs: Get "https://graph.microsoft.com/v1.0/users/***@******/drive/root/children?$top=5000&$expand=thumbnails($select=medium)&$select=id,name,size,lastModifiedDateTime,content.downloadUrl,file,parentReference": read tcp ***********09:54692->************72:443: wsarecv: An existing connection was forcibly closed by the remote host.这个是什么问题啊？

**Comment by hcllmsx** [comment]
OneDrive APP怎么挂载指定的目录啊，直接填写文件夹路径没有用诶

**Comment by liquidsnake9** [comment]
挂载成功，但上传文件速度极为缓慢，几十k，复制任务里还时不时得失败，至今一个上G得文件没上传成功过，为啥这么慢

**Reply by Gehinax** [reply]
确实，我pikpak复制到OneDrive，大文件没成功过，用的服务器搭建的

**Reply by Gehinax** [reply]
全是unexpected EOF

**Reply by Gehinax** [reply]
去添加存储界面改一下分片大小，把5改成100，服务器性能好的话200效率更好，上传大文件才能成功

**Comment by tianzecn** [comment]
请问支持世纪互联版吗

**Comment by SilveryLoong** [comment]
自定义HOST功能怎么用啊

**Comment by wkbenpao** [comment]
第 5 步授权的过程这样报错啥原因？
![Uploading 截屏2025-02-05 21.09.22.png…]()

**Comment by JusKyle** [comment]
根文件夹路径

这个怎么填 我只想挂载某个目录  不想挂载整个目录
